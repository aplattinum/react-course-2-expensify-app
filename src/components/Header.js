import React from 'react';
import {NavLink} from 'react-router-dom';
import { startLogout } from '../actions/auth';
import {connect} from 'react-redux';

export const Header = ({ startLogout }) => (
    <header className="header">
      <h1>Expensify</h1> 
      <div >
        <NavLink to="/dashboard" exact={true} activeClassName="is-active">Go Home</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
        <button onClick={startLogout} >Logout</button>  
      </div>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
}
)
export default connect(undefined, mapDispatchToProps )(Header);
