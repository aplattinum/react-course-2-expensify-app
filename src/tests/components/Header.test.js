import {shallow} from 'enzyme';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import Header from '../../components/Header';



test('should render header correctly', () => {
    const wrapper = shallow(<Header/>);
    expect(wrapper).toMatchSnapshot();
    
    
})