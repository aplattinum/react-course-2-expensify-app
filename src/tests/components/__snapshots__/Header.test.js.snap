// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header
  className="header"
>
  <h1>
    Expensify
  </h1>
  <div>
    <NavLink
      activeClassName="is-active"
      exact={true}
      to="/dashboard"
    >
      Go Home
    </NavLink>
    <NavLink
      activeClassName="is-active"
      to="/create"
    >
      Create Expense
    </NavLink>
    <button
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</header>
`;
